### ISSUE TITLE
Missing District data

### ISSUE THREAD

**Issue by resnape** [stwf/BMD-X#16]
The following districts are not being returned from the BMDDIST.TXT file. (Note:  These districts were contained in a 1964 transcription for marriages)

Cardigan S.,8a
Cambridge,4a
Durham C.,1a
E. Elloe,3b
King's L.,4b
Leicester C.,3a
Leighton B.,4a
Marylebone,5c
Middlesboro',1b
Middlesex S.,3a
Radnor E.,8c
St. Asaph,8a
St. Germans,7a
St. Helens,10f
St. Neots,4b
St. Pancras,5d
Stoke,9b
Sutton Coldf'ld,9c
W. Glamorgan,8b

Design Observation:  Remove the embedded BMDDIST.TXT, BMDTXT2.TXT and BMDNAME.TXT files so that it is not necessary to rebuilt MacBMD-X every time there is a change to the content of these files.

**Comment by stwf** [comment]
The town data is stored in the bmd_districts.txt file inside the app bundle. Items added here will be available after relaunching the app. BMDDIST.TXT and BMDDIST2.TXT are no longer used.

I will leave it to Peter Adams if this data  should be added.

**Comment by resnape** [comment]
Thanks.  I have figured out the logic of storing the District data, except for the 10th data item which does not appear to do anything in the code.  The 1971 Deaths districts have a great many differences and having to type in the District was slowing the process.  So I have, for my own edification and use, created an updated bmd_town_dump.txt file.  I have recorded the changes in a separate spreadsheet.  I shall shortly being transcribing 1974 births and I expect another raft of different districts.

**Comment by stwf** [comment]
ok, you could also make those changes in a branch of the code and make a pull request on that too. It would make it easier for Peter to review and incorporate your changes. Thanks

**Comment by stwf** [comment]
The changes you emailed to me have been put in this branch:
https://github.com/stwf/BMD-X/compare/data-fixes

They should be double checked and this can then be closed

**Comment by resnape** [comment]
As requested I have double checked the file.   There appears to be numerous entries after a new entry shows a deletion and insertion when there has been no change.  I am assuming that this is just an artefact with the system processing?

The following new entry appears to be missing.
        Here,1,1971,,,,,7c,7c,

Otherwise all is good

**Comment by stwf** [comment]
ok thanks. I added here! Also I think I cleaned up all of those false changes, I wasn't sure what that was. Thanks.

**Comment by archerbarrie** [comment]
I am concerned about the comment by resnape on 24th March 2015 that the code does not do anything with the 10th item in the district file. The last 5 items are the volumes for the 5 periods when a particular volume structure was used so the 10th item is the volume that should be used from 1st April 1974. If the 10th item is not being used then the volume will be wrong from that date. Furthermore the implication of the districts being different for 1971 is that the latest districts file is not being used - it is available at http://www.freebmd.org.uk/addons/winbmd/Districts.txt

**Comment by resnape** [comment]
Please ignore this observation.  I have subsequently deduced how the code accesses the 10th item in the district file.  The only other observation that I have regarding the district file is that there is an additional space between the district name and the preceding comma.  This manifests itself in a space in the transcription output in the same position.

**Comment by archerbarrie** [comment]
Thanks. The district file contains four districts with commas:
Bradford, W.
Newport, Sh.
Richmond, S.
Whitchurch, S.
In the resulting transcription these have to be enclosed in quotes (since the fields are comma delimited). The standard district file does not have an extra space as described. So is the issue that an extra space have been inserted before the comma in the resulting transcription? Or does every district have an extra space after the preceding (field separating) comma in the transcription? Or is a non-standard district file being used?

**Comment by resnape** [comment]
Apologies I was not very succinct in my explanation.
This is a sample of the current districts file used in the latest MacBMD-X (1.0.9)

Abingdon,,,3,2001,6,2c,6a,6a,20 
Acle,1,1939,1,1974,,4b,4b,4b, 
Alcester,,,3,1985,16,6d,9c,9c,31 

After the 10th entry  20 & 31 there is a hard space which manifests in the following:

SNEDDON,JENNIFER J.,WAKEFIELD,ABINGDON,20 ,0521
SNEDDON,PATRICIA A. E.,BROWN,ALCESTER,31 ,0125S

i.e. a space after the district numbers 20 & 31.

I have resolved this by using a locally modified districts file but I am looking at a low priority software suggestion for Steve to consider to cater for such spaces

**Comment by stwf** [comment]
Yes, I can absolutely clear this up. Let me know if there are any other issues that need to be cleared up.
